sr_unix/gtm_unique_file_util.c

```diff
@@ -16,32 +19,41 @@
 #include "iosp.h"		/* for SS_NORMAL */
 
 /* Checks whether the two fileids passed are identical  */
-xc_status_t gtm_is_file_identical(xc_fileid_ptr_t fileid1, xc_fileid_ptr_t fileid2)
+ydb_status_t gtm_is_file_identical(ydb_fileid_ptr_t fileid1, ydb_fileid_ptr_t fileid2)
 {
-	if (!fileid1 || (!fileid2)) return FALSE;
+	if (!fileid1 || (!fileid2))
+		return FALSE;
 	return is_gdid_identical((gd_id_ptr_t) fileid1, (gd_id_ptr_t) fileid2);
 }
 
 /* Converts given filename to unique file id. Uses filename_to_id internally for getting the unique id. Note that,
  * the allocation of the fileid structure is done here and the caller needs to worry only about free'ing the
  * allocated pointer via gtm_xcfileid_free. */
-xc_status_t gtm_filename_to_id(xc_string_t *filename, xc_fileid_ptr_t *fileid)
+ydb_status_t gtm_filename_to_id(ydb_string_t *filename, ydb_fileid_ptr_t *fileid)
 {
+	int		actstatus;
 	boolean_t	status;
 	gd_id_ptr_t	tmp_fileid;
 
 	if (!filename)
 		return FALSE;
 	assert(fileid && !*fileid);
 	tmp_fileid = (gd_id_ptr_t)malloc(SIZEOF(gd_id));
-	status = (SS_NORMAL == filename_to_id(tmp_fileid, filename->address));
-	*fileid = (xc_fileid_ptr_t)tmp_fileid;
-	return status;
+	actstatus = filename_to_id(tmp_fileid, filename->address);
+	status = (SS_NORMAL == actstatus);
+	if (status)
+		*fileid = (ydb_fileid_ptr_t)tmp_fileid;
+	else
+	{	/* There was an error */
+		free(tmp_fileid);
+		*fileid = NULL;
+	}
+	return status ? YDB_OK : actstatus;
```

Comment:
I have not thought much about this but seems like not changing the gtm_* flavors of the functions would be an easy change. In any case, I think you will own this issue.

Reply:
Here is the description of ydb_file_name_to_id() which is a wrapper for gtm_filename_to_id():

ydb_file_name_to_id() returns YDB_OK **_or an error return code._**

We can not do that hilighted part because the original version of gtm_filename_to_id() does not do return codes. It returns 1 if it was successful and 0 if it wasn't but does not return any information. Any attempt to rework this routine to return error information is equivalent to returning "TRUE" when it isn't. I need to revert this routine and we can only return success/failure. The spec needs to change for ydb_file_name_to_id() because of this.